psl-core/src/main/java/org/linqs/psl/application/inference/online/OnlineInference.java

```diff
@@ -0,0 +1,309 @@
+    }
+
+    /**
+     * Get objects ready for inference.
+     * This will call into the abstract method completeInitialize().
+     */
+    @Override
+    protected void initialize() {
+        startServer();
+
+        log.debug("Creating persisted atom manager.");
+        atomManager = createAtomManager(db);
+        log.debug("Atom manager initialization complete.");
+
+        initializeAtoms();
+
+        reasoner = createReasoner();
+        termStore = createTermStore();
+        groundRuleStore = createGroundRuleStore();
+        termGenerator = createTermGenerator();
+
+        int atomCapacity = atomManager.getCachedRVACount() + atomManager.getCachedOBSCount();
+        termStore.ensureVariableCapacity(atomCapacity);
+
+        if (normalizeWeights) {
+            normalizeWeights();
+        }
+
+        if (relaxHardConstraints) {
+            relaxHardConstraints();
+        }
+
+        close = false;
+        objective = 0;
+
+        completeInitialize();
+    }
+
+    @Override
+    protected OnlineTermStore createTermStore() {
+        return (OnlineTermStore)Options.INFERENCE_TS.getNewObject();
+    }
+
+    @Override
+    protected OnlineAtomManager createAtomManager(Database db) {
+        return new OnlineAtomManager(db);
+    }
+
+    private void startServer() {
+        try {
+            server = new OnlineServer<OnlineAction>();
+            server.start();
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> server.closeServer()));
+        } catch (IOException e) {
+            log.info("Failed to start server");
+            close();
+            System.exit(1);
+        }
+    }
+
+    protected void executeAction(OnlineAction nextAction) throws IllegalArgumentException {
+        if (nextAction.getClass() == UpdateObservation.class) {
+            doUpdateObservation((UpdateObservation)nextAction);
+        } else if (nextAction.getClass() == AddAtom.class) {
+            doAddAtom((AddAtom)nextAction);
+        } else if (nextAction.getClass() == DeleteAtom.class) {
+            doDeleteAtom((DeleteAtom)nextAction);
+        } else if (nextAction.getClass() == WriteInferredPredicates.class) {
+            doWriteInferredPredicates((WriteInferredPredicates)nextAction);
+        } else if (nextAction.getClass() == Close.class) {
+            doClose((Close)nextAction);
+        } else {
+            throw new IllegalArgumentException("Action: " + nextAction.getClass().getName() + " Not Supported.");
+        }
+    }
+
+    protected void doAddAtom(AddAtom nextAction) throws IllegalArgumentException {
+        // Resolve Predicate
+        Predicate registeredPredicate = Predicate.get(nextAction.getPredicateName());
+        if (registeredPredicate == null) {
+            throw new IllegalArgumentException("Predicate is not registered: " + nextAction.getPredicateName());
+        }
+
+        switch (nextAction.getPartitionName()) {
+            case "READ":
```

Comment:
Use a config option to set a consistent read partition that we will add to.
If specified, use that.
Else (default case), use the first read partition in the db.